src/wmtk/SimplicialComplex.cpp

```diff
@@ -438,46 +420,41 @@ bool SimplicialComplex::edge_collapse_possible_2d(Tuple t, const Mesh& m)
     // assert(opp(h1) != next(h1))
 
     auto h0 = next(t);
-    if (!m.is_boundary(h0))
-    {
-        if (opp(h0) == next(h0))
-        {
+    if (!m.is_boundary(h0)) {
+        if (opp(h0) == next(h0)) {
             return false;
         }
     }
 
-    if (!m.is_boundary(t))
-    {
+    if (!m.is_boundary(t)) {
         auto h1 = next(opp(t));
-        if (!m.is_boundary(h1))
```

Comment:
This logic with h1 is the same as before with t? Why not use a single function to avoid duplicati code

Reply:
I refactored the `edge_collapse_possible_2d` function completely. It should be fine now.